lib/helpers/getDomNodeProfile.js

```diff
@@ -13,23 +15,31 @@ const getTextContent = (el, options) => {
 }
 
 export function getDomNodeProfile(el, options = {}) {
-  return {
-    action: el.action,
+  let profile = {
     class: el.className,
     href: el.href || null,
     id: el.id,
-    method: el.method,
     name: el.name,
-    ng_click: el.getAttribute('ng-click'),
-    ng_model: el.getAttribute('ng-model'),
+    all_attrs:  getDomNodeAttributes(el, /^((?!value).)*$/),
     node_name: el.nodeName,
-    selector: getDomNodePath(el),
     tag_name: el.tagName,
     text: el.text,
-    text_content: getTextContent(el, options),
     title: el.title,
-    type: el.type,
-    x_position: el.offsetLeft || el.clientLeft || null,
-    y_position: el.offsetTop || el.clientTop || null
+    type: el.type
   };
+
+  if (!options.basicProfile) {
+    profile = {
+      ...profile,
+      action: el.action,
+      method: el.method,
+      n_parents: getNParents(el, options.nParents),
```

Comment:
It is rather confusing to recurse from A -> B -> B -> B -> B -> A -> A -> A. Is there a simpler way to walk up the tree and get a profile of each node? If needed, we can always flatten later. 

I fear someone coming in later and changing the options of these methods in a way that recurses infinitely. It's not really obvious `getNParents` is going to call `getDomNodeProfile` a bunch more times.

Reply:
I'm not sure I follow your concern here; are you worried that at some point, `getNParents` could be moved into the `basicProfile` logic, and we wind up in a never-ending loop?